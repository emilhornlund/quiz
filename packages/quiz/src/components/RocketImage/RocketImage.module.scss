@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: row;
  justify-content: center;

  > img {
    @include mobile {
      width: 140px;
      height: 140px;
    }

    @include tablet {
      width: 170px;
      height: 170px;
    }

    @include desktop {
      width: 200px;
      height: 200px;
    }
  }
}
