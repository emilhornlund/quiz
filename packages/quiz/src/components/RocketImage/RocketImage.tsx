import React, { FC } from 'react'

import Rocket from '../../assets/images/rocket.svg'

import styles from './RocketImage.module.scss'

const RocketImage: FC = () => (
  <div className={styles.main}>
    <img src={Rocket} alt="rocket" />
  </div>
)

export default RocketImage
