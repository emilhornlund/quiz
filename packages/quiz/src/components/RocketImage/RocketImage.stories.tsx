import type { Meta, StoryObj } from '@storybook/react'

import RocketImage from './RocketImage'

const meta = {
  component: RocketImage,
} satisfies Meta<typeof RocketImage>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {},
} satisfies Story
