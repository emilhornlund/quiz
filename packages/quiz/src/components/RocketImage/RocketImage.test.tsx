import { render } from '@testing-library/react'
import React from 'react'
import { expect, test } from 'vitest'

import RocketImage from './RocketImage'

test('should render RocketImage with default props', async () => {
  const { container } = render(<RocketImage />)

  expect(container).toMatchSnapshot()
})
