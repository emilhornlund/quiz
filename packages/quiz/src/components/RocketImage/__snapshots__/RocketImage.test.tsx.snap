// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render RocketImage with default props 1`] = `
<div>
  <div
    class="main"
  >
    <img
      alt="rocket"
      src="/src/assets/images/rocket.svg"
    />
  </div>
</div>
`;
